rfcs/text/0014_notification_service.md

```diff
@@ -0,0 +1,373 @@
+```
+
+Writing to the same Notification storage might cause conflict on the `write`. If the storage mechanism implementation
+doesn’t make them impossible, we will have to use a single place to change notification status (via the [queue](#localrepository)).
+
+## Settings
+Provides HTTP API for Kibana UI to configure user-specific settings.
+Not provided in MVP, we might add in the future:
+- Unsubscribe from a source
+- Subscribe to a source
+
+#### Filtering
+Filtering allows users to hide notifications they aren’t interested in. Kibana UI attaches a list of filters when requesting notifications.
+Kibana stores filter as a part of User settings data to provide a seamless experience between different deployments on Cloud and user sessions.
+
+
+# Drawbacks
+
+- It is not entirely clear what functionality will be supported by UNS and when this will be known.
+It forces us to evaluate each feature separately as to how difficult it will be to adapt it to UNS later.
+- The service relies on the yet-to-be-added ability to uniquely identify the user, enumerate users with a given set of permissions.
+- The service doesn't provide a built-in security mechanism but relies on a notification source implementation.
+- Lack of a centralized i18n service makes message transferring over-bloated due to the necessity to send all the possible translations over the wire.
+
+
+# Alternatives
+
+### Threat Cloud as another pluggable storage mechanism.
+Kibana notification service collects internal notifications and shows them in the UI for both on-prem and Cloud environments.
+Kibana reuses Notification service implementation on Cloud but enhances it with additional pluggable storage of external notifications.
+![img](../images/0014/alternative_implementation.png)
+
+The main benefit of this approach is that Kibana always uses the same service in any environment.
+But in addition to the restrictions listed in the [Drawbacks](#drawbacks) section, the following are added:
+- Each retrieval of a notification list still requires waiting for a roundtrip to UNS.
+- The list of Kibana notifications stored in an instance might be out of sync with a copy stored in UNS due to network problems. 
+- Storing notifications in the Cloud deployment and their copy in UNS consumes twice the disk space.
+
+# Adoption strategy
+
+We start with introducing Kibana Notification Service as a Kibana plugin, adding more notification sources along the road.
+Integration with UNS is something that is not going to happen any time soon. The work can be divided into the following phases:
+
+### Phase I
+Kibana shows notifications created by the Newsfeed Kibana plugin in Notification flyout.
+![img](../images/0014/kns_flyout.png)
+A notification cannot have a particular recipient in the lack of a way to identify a user in the system. A notification will be shown to all the users instead.
+User-specific notification state (*read_status*) is not stored in Kibana but browser Local Storage.
```

Comment:
Are we planning on adding a migration between phase 1 and 2 to move back the user's notification state from local storage to the backend, or are we good just not handling that?

Reply:
The way I see it that we can neglect migration. There is a high chance that a user will see some notifications as `new` several times during a release. Because `local storage` is not a reliable source of information: 
- it can be unavailable
- it can be cleared
- it can be full
- Kibana cannot uniquely identify a user profile.

Another alternative is not to support read / unread status until we have a user-specific data storage mechanism.

Reply:
> Another alternative is not to support read / unread status until we have a user-specific data storage mechanism.

Honestly, I lean towards this option. It allows to avoid some work now that will be thrown away while also eliminating a confusion situation for the user when logging in from a different browser  or device.

Reply:
Fair enough. Removed the state from the `Phase I`